pkg/ipam/types/types.go

```diff
@@ -159,6 +209,22 @@ type IPAMStatus struct {
 	ReleaseIPs map[string]IPReleaseStatus `json:"release-ips,omitempty"`
 }
 
+// IPAMPoolRequest is a request from the agent to the operator, indicating how
+// may IPs it requires from a given pool
+type IPAMPoolDemand struct {
+	// IPv4Addrs contains the number of requested IPv4 addresses out of a given
+	// pool
+	//
+	// +optional
+	IPv4Addrs int `json:"ipv4-addrs,omitempty"`
```

Comment:
yeah, camelCase is the convention for k8s yaml / json documents.

Reply:
Almost all other CiliumNode fields are also `kebab-case` unfortunately, see for example the ENISpec used to manage the ENI IPAM mode: https://github.com/cilium/cilium/blob/c5cbf403dbe355ecbb80dfc8d7a8ed4da45c43bd/pkg/aws/eni/types/types.go#L18

I made `ipv4-addrs` and `ipv6-addrs` kebab-case to be consistent with the existing IPAM fields.

If you feel strongly that we should make newer fields camelCase, I can change it, but otherwise I'll leave it kebab-case.

Reply:
Oh that's unfortunate. In that case, better to be consistent with kebab-case.